Clarify what profiling OID4VP means (#339)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

Any of the OAuth 2.0 related specifications, such as [@RFC9126] and [@RFC9101], and Best Current Practice (BCP) documents, such as [@RFC8252] and [@I-D.ietf-oauth-security-topics], can be implemented on top of this specification.

> In summary, OpenID for Verifiable Presentations is a framework that requires profiling

Reviewer: Shouldn't this text also mention that a profile needs to select a certain credential format?

Author: yes, addressed below.

> to achieve interoperability. Out of the

Reviewer 2: This text is quite generic and I'm not sure if I would agree with the phrasing. The issue I have with the phrasing is that a profile can also have optional parameters, not only mandatory ones.

IMO, a profile does the following for a specific use case or ecosystem:
- define choices of values for mandatory and optional features, e.g., allowed credential format identifiers.
- define what optional features do not apply or mandate optional features
- optionally extend OID4VP if new features are required

This description is still vague but I could probably live with this better.

However, I think it would be better to define concretely what exactly has to be further defined by a profile.

> to achieve interoperability within a certain ecosystem or a jurisdiction. Out of the

Author: added HAIP lower: https://github.com/openid/OpenID4VP/pull/339/files#r1851849652
but maybe there is a better place to put HAIP..

> In summary, OpenID for Verifiable Presentations is a framework that requires profiling
> to achieve interoperability. Out of the
> features that are optional in this specification, a profile MUST select features that are
> mandatory and define a set of specific requirements.

Author: ```suggestion
In summary, OpenID for Verifiable Presentations is a framework that requires profiling
to achieve interoperability. Profiling means defining:
- what optional features are used or mandatory to implement; e.g., response encryption.
- which values are permitted for parameters, e.g., credential format identifiers.
- optionally, extensions for new features
```

MUST perform its own security checks on the returned Credentials and
presentations.

> ## Credential Format Specific Parameters {#format_specific_properties}
> 
> OpenID for Verifiable Presentations is Credential Format agnostic, i.e., it is designed to allow applications to request and receive Verifiable Presentations and Verifiable Credentials in any Credential Format. This section defines sets of Credential Format specific parameters for some of the known Credential Formats. Other specifications or deployments can define their own set of Credential Format specific parameters:
> 
> ### IETF SD-JWT VC {#format_vc_sd_jwt}
> 
> `vct_values`:
> : OPTIONAL. An array of strings that specifies allowed values for
> the type of the requested Verifiable Credential. All elements in the array MUST
> be valid type identifiers as defined in [@!I-D.ietf-oauth-sd-jwt-vc]. The Wallet
> MAY return credentials that inherit from any of the specified types, following
> the inheritance logic defined in [@!I-D.ietf-oauth-sd-jwt-vc].
> 
> ### Mobile Documents or mdocs (ISO/IEC 18013 and ISO/IEC 23220 series) {#format_mso_mdoc}
> 
> `doctype_value`:
> : OPTIONAL. String that specifies an allowed value for the
> doctype of the requested Verifiable Credential. It MUST
> be a valid doctype identifier as defined in [@ISO.18013-5].
> 

Author: ```suggestion
```

## Static Configuration Values of the Wallets

This document lists profiles of this specification that define static configuration values of the Wallets and defines one set of static configuration values that can be used by the Verifier when it is unable to perform Dynamic Discovery and is not using any of the profiles.

### Profiles that Define Static Configuration Values

The following is a list of profiles that define static configuration values of Wallets:

> - [JWT VC Presentation Profile](https://identity.foundation/jwt-vc-presentation-profile/)

Author: ```suggestion
- [OpenID4VC High Assurance Interoperability Profile](https://openid.github.io/oid4vc-haip-sd-jwt-vc/openid4vc-high-assurance-interoperability-profile-sd-jwt-vc-wg-draft.html)
- [JWT VC Presentation Profile](https://identity.foundation/jwt-vc-presentation-profile/)
```
